### PR TITLE
[ffigen] rename assetId to asset-id

### PR DESCRIPTION
**Pull request by TheComputerM**
Rename the config option under `ffi-native` to follow dash-case for uniformity with the other config options.

---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/blob/main/docs/External-Package-Maintenance.md#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

**Comment by dcharkes** [comment]
Should we have a grace period where the old key is still accepted but throws a warning/error in the console? (But we do remove the key from the schema so that it gives an error in people's IDEs.)

**Comment by TheComputerM** [comment]
Good idea

File: pkgs/ffigen/lib/src/config_provider/spec_utils.dart [hunk 1]
```diff
@@ -627,6 +627,18 @@ StructPackingOverride structPackingOverrideExtractor(
 
 FfiNativeConfig ffiNativeExtractor(dynamic yamlConfig) {
   final yamlMap = yamlConfig as Map?;
+
+  // Use the old 'assetId' key if present but give a deprecation warning
+  if (yamlMap != null &&
+      !yamlMap.containsKey(strings.ffiNativeAsset) &&
+      yamlMap.containsKey('assetId')) {
+    _logger.warning("DEPRECATION WARNING: use 'asset-id' instead of 'assetId'");
```
Comment (dcharkes) [inline pkgs/ffigen/lib/src/config_provider/spec_utils.dart:635]: Should this be covered by a test?

Comment (TheComputerM) [reply pkgs/ffigen/lib/src/config_provider/spec_utils.dart:635]: Done.

**Review by dcharkes — APPROVED** [review]

**Comment by TheComputerM** [comment]
Forgot to regen ffigen.schema.json 😅

**Pull request merged**